processor/src/main/java/org/mapstruct/ap/internal/model/BeanMappingMethod.java

```diff
@@ -1636,11 +1636,11 @@ private ReportingPolicyGem getUnmappedTargetPolicy() {
             if ( mappingReferences.isForForgedMethods() ) {
                 return ReportingPolicyGem.IGNORE;
             }
-            // If we have ignoreByDefault = true, unprocessed target properties are not an issue.
-            if ( method.getOptions().getBeanMapping().isignoreByDefault() ) {
```

Comment:
Why is this changed needed? This method is for the unmapped target policy and you are adding functionality for the unmapped source policy. Perhaps the null check on line 1639 is obsolete, but let's try to keep the PRs focused on the main functionality and leave things like this for other PRs